src/ImageSharp/Common/Memory/PixelDataPool{T}.cs

```diff
@@ -9,39 +9,69 @@ namespace ImageSharp
     using System.Buffers;
 
     /// <summary>
-    /// Provides a resource pool that enables reusing instances of value type arrays <see cref="T:T[]"/>.
-    /// <see cref="Rent(int)"/> will always return arrays initialized with 'default(T)'
+    /// Provides a resource pool that enables reusing instances of value type arrays for image data <see cref="T:T[]"/>.
     /// </summary>
     /// <typeparam name="T">The value type.</typeparam>
-    public static class PixelDataPool<T>
+    public class PixelDataPool<T>
         where T : struct
     {
         /// <summary>
-        /// The <see cref="ArrayPool{T}"/> used to pool data.
+        /// The <see cref="ArrayPool{T}"/> which will be always cleared.
         /// </summary>
-        private static readonly ArrayPool<T> ArrayPool = ArrayPool<T>.Create(CalculateMaxArrayLength(), 50);
+        private static readonly ArrayPool<T> CleanPool = ArrayPool<T>.Create(CalculateMaxArrayLength(), 50);
+
+        /// <summary>
+        /// The <see cref="ArrayPool{T}"/> which is not kept clean.
+        /// </summary>
+        private static readonly ArrayPool<T> DirtyPool = ArrayPool<T>.Create(CalculateMaxArrayLength(), 50);
+
+        /// <summary>
+        /// The backing <see cref="ArrayPool{T}"/>
+        /// </summary>
+        private ArrayPool<T> arrayPool;
+
+        /// <summary>
+        /// A value indicating whether clearArray is requested on <see cref="ArrayPool{T}.Return(T[], bool)"/>.
+        /// </summary>
+        private bool clearArray;
+
+        private PixelDataPool(ArrayPool<T> arrayPool, bool clearArray)
+        {
+            this.clearArray = clearArray;
+            this.arrayPool = arrayPool;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="PixelDataPool{T}"/> which will always return arrays initialized to default(T)
+        /// </summary>
+        public static PixelDataPool<T> Clean { get; } = new PixelDataPool<T>(CleanPool, true);
```

Comment:
Okkay, it's a static class again, no longer clearing, we can call explicit clear methods instead:
- `ImageBase<T>.ClearPixels()`
- `PinnedBuffer<T>.Clear()`
- `ArrayPointer<T>.Clear(int)`